pkg/controller/portworxdiag/portworxdiag.go

```diff
@@ -231,13 +232,84 @@ func getNodeIDToStatusMap(nodeStatuses []diagv1.NodeStatus) map[string]string {
 	return statuses
 }
 
+func (c *Controller) getNodeIDsWithSelectedVolumes(diag *diagv1.PortworxDiag, stc *corev1.StorageCluster) ([]string, error) {
+	if diag == nil || diag.Spec.Portworx == nil || diag.Spec.Portworx.VolumeSelector.IDs == nil || diag.Spec.Portworx.VolumeSelector.Labels == nil {
+		return []string{}, nil
+	}
+
+	nodeIDMap := map[string]bool{}
+
+	volumeClient := api.NewOpenStorageVolumeClient(c.grpcConn)
+	ctx, err := pxutil.SetupContextWithToken(context.Background(), stc, c.client)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, id := range diag.Spec.Portworx.VolumeSelector.IDs {
+
+		// Inspect volume to get location of replicas
+		volumeInspectResponse, err := volumeClient.Inspect(
+			ctx,
+			&api.SdkVolumeInspectRequest{
+				VolumeId: id,
+			},
+		)
+		if err != nil {
+			logrus.Warnf("Failed to inspect volume %s: %v", id, err)
+			// Try the next volume
+			continue
+		}
+
+		// Add all nodes that have a replica of this volume
+		for _, rs := range volumeInspectResponse.Volume.ReplicaSets {
+			for _, r := range rs.Nodes {
+				nodeIDMap[r] = true
+			}
+		}
+	}
+
+	// Inspect all volumes by labels
+	if len(diag.Spec.Portworx.VolumeSelector.Labels) > 0 {
+		volumeInspectResponse, err := volumeClient.InspectWithFilters(
+			ctx,
+			&api.SdkVolumeInspectWithFiltersRequest{
+				Labels: diag.Spec.Portworx.VolumeSelector.Labels,
+				Options: &api.VolumeInspectOptions{
+					Deep: true,
```

Comment:
Did we verify a deep inspect is needed in order to see ReplicaSets?

Deep inspecting all volumes is a very expensive operation and has caused system slowness

Reply:
My initial pass at the code it seemed like it, but I'm not 100% sure. I can try without it and see.

Reply:
.... given I accidentally didn't include it for single volumes and it still worked, apparently not! I'll remove that